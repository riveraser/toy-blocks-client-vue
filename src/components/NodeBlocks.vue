<template>
  <div class="node-block">
    <h2>{{id.padStart(3, '0')}}</h2>
    <div class="block-content">
      {{content}}
    </div>
  </div>
</template>

<script>
export default {
  name: "NodeBlocks",
  props: {
      id: String,
      content: String
  }
};
</script>
<style lang="scss">
@import '@/scss/_variables';

.node-block {
  /* Disabled */
  background: rgba(0, 0, 0, 0.12);
  border-radius: 2px;
  padding: 8px;
  font-family: Roboto;
  font-style: normal;
  font-weight: normal;
  margin-bottom: 4px;

  h2 {
    font-size: 10px;
    line-height: 16px;
    /* identical to box height, or 160% */
    letter-spacing: 1.5px;
    text-transform: uppercase;

    /* Primary Color */
    color: $primaryColor;
  }
  .block-content {
    /* 14 sp â€¢ Body 2 */

    font-size: 14px;
    line-height: 20px;
    /* identical to box height, or 143% */
    letter-spacing: 0.25px;

    /* Dark Color */
    color: $darkColor;
  }
}
</style>
