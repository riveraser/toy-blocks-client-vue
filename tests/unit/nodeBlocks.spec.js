import Vue from 'vue';
import { shallowMount } from '@vue/test-utils'
import NodeBlocks from '@/components/NodeBlocks.vue'
import Vuetify from 'vuetify'

Vue.use(Vuetify)

describe('<NodeBlocks />', () => {
  const defaultProps = {
    id:"1",
    content:"Duis mollis, est non commodo luctus, nisi erat porttitor ligula, eget lacinia odio sem nec elit."
  };

   //mocks fetch API
  
  //  global.fetch = jest.fn(() =>
  //   Promise.resolve({
  //     json: () => Promise.resolve({ url: 'https://thawing-springs-53971.herokuapp.com' }, data: [{},{}]  }),
  //   })
  // );

  const render = (props = {}) => shallowMount(NodeBlocks, {
    propsData: {
      ...defaultProps,
      ...props
    }
  });
  
  it('renders', () => {
    const wrapper = render();
    expect(wrapper.find('.block-content').exists()).toBeTruthy;
  });

  it('match the snapshot', () => {
    const wrapper = render();
    expect(wrapper).toMatchSnapshot();
  });
  
});