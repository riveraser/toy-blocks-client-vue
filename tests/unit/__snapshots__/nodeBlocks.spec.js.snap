// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NodeBlocks /> match the snapshot 1`] = `
<div class="node-block">
  <h2>001</h2>
  <div class="block-content">
    Duis mollis, est non commodo luctus, nisi erat porttitor ligula, eget lacinia odio sem nec elit.
  </div>
</div>
`;
